prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -795,3 +775,78 @@ impl<'tcx, 'vir: 'enc, 'enc> Enc<'tcx, 'vir, 'enc>
         }
     }
 }
+
+mod rev_doms {
+    /// Identical to `body.basic_blocks.dominators()` except in reverse. Since
+    /// there may be multiple `Return`/`Unreachable`/etc. terminators, we add a
+    /// special end block index which is invalid in `basic_blocks` but pretends
+    /// to be the successor of all these no-successor blocks.
+    pub struct ReverseDominators {
+        pub dom: dominators::Dominators<mir::BasicBlock>,
+        pub end: mir::BasicBlock,
+    }
+    impl ReverseDominators {
+        pub fn new<'a, 'tcx>(blocks: &'a mir::BasicBlocks<'tcx>) -> Self {
+            let no_succ_blocks = blocks.iter_enumerated().filter(|(_, data)|
+                data.terminator().successors().next().is_none()
+            ).map(|(bb, _)| bb).collect();
+            let rbb = RevBasicBlocks(blocks, no_succ_blocks);
+            Self {
+                dom: dominators::dominators(&rbb),
+                end: rbb.start_node(),
+            }
+        }
+        pub fn join_point(&self, bb: mir::BasicBlock) -> mir::BasicBlock {
```

Comment:
Even though this module is private, it is about reverse dominators, not join points, so I would rename this.